main/src/ca/uwaterloo/flix/language/ast/ResolvedAst.scala

```diff
@@ -25,13 +25,16 @@ import scala.collection.immutable.List
 
 object ResolvedAst {
 
-  case class Root(defs: Map[Symbol.DefnSym, ResolvedAst.Def],
+  case class Root(classes: Map[Symbol.ClassSym, ResolvedAst.Class],
+                  defs: Map[Symbol.DefnSym, ResolvedAst.Def],
                   enums: Map[Symbol.EnumSym, ResolvedAst.Enum],
                   latticeOps: Map[Type, ResolvedAst.LatticeOps],
                   properties: List[ResolvedAst.Property],
                   reachable: Set[Symbol.DefnSym],
                   sources: Map[Source, SourceLocation])
 
+  case class Class(sym: Symbol.ClassSym, tparams: List[ResolvedAst.TypeParam], signatures: List[ResolvedAst.Sig], superclasses: List[ResolvedAst.Class])
```

Comment:
I would like to take a very iterative approach (because I believe there will be plenty of complexity). So I suggest that we drop the super classes and only have one tparams.

Reply:
Oh, can I ask you to move Class just before Sig that way we can keep them "together".